ssl/pom.xml

```diff
@@ -81,6 +81,14 @@
             <scope>provided</scope>
         </dependency>
 
+        <!-- https://mvnrepository.com/artifact/com.google.code.gson/gson -->
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.6</version>
```

Comment:
In general if we are adding a new dependency that dependency needs to be defined in the root pom of the project with the version specified as a property to sub modules can use the previously defined dependency.

However if this is just for a JSON value do we not already have something in our dependencies we could use instead?

If we do need a new dependency we should also check if this dependency is already available in WildFly Core or WildFly as there may be an overhead bringing in the dependency there.

Reply:
@darranl Thank you. Do you think the approach in this PR is good? I used gson here for serialization / deserialization for the sole purpose of copying the SSLParameters that were added in java 9 or later (because their setter methods do not exist in java 8). 

Gson does this by converting a Java object to its JSON representation and back. However this has a disadvantage of being slow. I was thinking of other possible solutions that I mentioned in this [comment](https://github.com/wildfly-security/wildfly-elytron/pull/1384#discussion_r408037728). But if this approach is OK I will check out wildfly and wildfly-core dependencies and update the PR.